internal/bootloader/bootloader.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package bootloader
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Bootloader provides primitives to retain state across boots and handle
+// boot slots.
+type Bootloader interface {
+	// Name returns the bootloader name.
+	Name() string
+
+	// GetBootVars populates the specified variables from the bootloader.
+	// These variables are preserved across reboots.
+	GetBootVars(names ...string) (map[string]string, error)
```

Comment:
In the end, we decided to get rid of these `GetBootVars`/`SetBootVars` methods altogether, since there is no requirement for those anymore (we just need the `Status()`, and `ActiveSlot()`/`SetActiveSlot()` for interacting with the boot environment)